src/hooks/useLoader.js

```diff
@@ -0,0 +1,38 @@
+import path from 'path'
+import { useState } from 'react'
+
+export const STATUS = {
+  idle: 'idle',
+  loading: 'loading',
+  done: 'done',
+}
+
+export function useLoadMore ({ initialPageInfo, initialData }) {
+  const [currentPage, setCurrentPage] = useState(1)
+  const [data, setData] = useState(initialData)
+  const [pageInfo, setPageInfo] = useState(initialPageInfo)
+  const [status, setStatus] = useState(STATUS.idle)
+
+  const loadNextPage = async () => {
+    const nextPage = currentPage + 1
+    setCurrentPage(nextPage)
+    setStatus(STATUS.loading)
+
+    const files = path.join('/', `${nextPage}.json`)
+
+    const response = await fetch(files)
+    const data = await response.json()
+
+    setData(state => [...state, ...data.nodes])
+    setPageInfo(() => data.pageInfo)
+    setStatus(() => STATUS.done)
+  }
```

Comment:
O último comentário vale para este caso, o `useMemo` retorna um valor memoizado. Neste caso como é uma função de callback é recomendado o uso do `useCallback` para declarar esta função dentro do hook :)
- Retira os callbacks (`() =>`) sem `state` como parâmetro, passando o valor diretamente para as funções _handlers_ é a mesma coisa que isto.
- Desestrutura o retorno de `response.json()`, fica mais agradável :)

https://reactjs.org/docs/hooks-reference.html#usecallback

```suggestion

  const loadNextPage = useCallback(async () => {
    const nextPage = currentPage + 1
    setCurrentPage(nextPage)
    setStatus(STATUS.loading)

    const files = path.join('/', `${nextPage}.json`)

    const response = await fetch(files)
    const {pageInfo, nodes} = await response.json()

    setData(state => [...state, ...nodes])
    setPageInfo(pageInfo)
    setStatus(STATUS.done)
  }, [currentPage])
```